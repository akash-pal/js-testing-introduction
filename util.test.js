const {generateText, checkAndGenerate} =  require('./util');

/* unit test*/
test('should output name and age', () => {
    const text = generateText('Akash', 27);
    expect(text).toBe('Akash (27 years old)');
    const text2 = generateText('Max', 27);
    expect(text2).toBe('Max (27 years old)');
});

test('should outut data-less text', () => {
    const text = generateText('', null);
    expect(text).toBe(' (null years old)')
});

/*intergration test*/
test('should generate a valid text output', () => {
    const text = checkAndGenerate('Akash', 27);
    expect(text).toBe('Akash (27 years old)');
});