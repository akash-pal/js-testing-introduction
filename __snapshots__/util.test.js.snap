// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking the format 1`] = `"Akash (27 years old)"`;
